Recast/Include/RecastAlloc.h

```diff
@@ -214,19 +214,28 @@ void rcVectorBase<T, H>::resize_impl(rcSizeType size, const T* value) {
 		destroy_range(size, m_size);
 		m_size = size;
 	} else if (size > m_size) {
-		T* new_data = allocate_and_copy(size);
-		// We defer deconstructing/freeing old data until after constructing
-		// new elements in case "value" is there.
-		if (value) {
-			construct_range(new_data + m_size, new_data + size, *value);
+		if (size <= m_cap) {
+			if (value) {
+				construct_range(m_data + m_size, m_data + size, *value);
+			} else {
+				construct_range(m_data + m_size, m_data + size);
+			}
+			m_size = size;
 		} else {
-			construct_range(new_data + m_size, new_data + size);
+			T* new_data = allocate_and_copy(size);
+			// We defer deconstructing/freeing old data until after constructing
+			// new elements in case "value" is there.
+			if (value) {
+				construct_range(new_data + m_size, new_data + size, *value);
+			} else {
+				construct_range(new_data + m_size, new_data + size);
+			}
+			destroy_range(0, m_size);
+			rcFree(m_data);
+			m_data = new_data;
+			m_cap = size;
+			m_size = size;
```

Comment:
As @Bromeon points out, this strategy is still very inefficient for `push3` since it will repeatedly allocate on every `push3`. Instead, we should grow it by doubling the capacity until it is larger than `n`. The old implementation before #318 was https://github.com/recastnavigation/recastnavigation/blob/18562383f4c5cffa0678c709049340516ebc5e40/Recast/Source/RecastAlloc.cpp#L76-L85.
Can you do something similar here?

Reply:
I agree with this, it's how I fixed inefficiencies in wildmidi. Each time we had to realloc, we just doubled the size. As th saying goes, RAM is cheap.

Reply:
Done